### PR TITLE
Fix Counter redirecting Encored self-target moves

### PR DESCRIPTION
**Pull request by pacmanboss256**
Counter/mirror coat target redirection carried over because of the decreased priority, this fix makes it so the target is not redirected if the move being used is not counter or mirror coat.

### CODE REVIEW

**Comment by pacmanboss256** [comment]
referenced here https://github.com/smogon/pokemon-showdown/projects/3#card-52058554

**Comment by Zarel** [comment]
This seems like the wrong place to fix it? `OverrideAction` should probably be the place that re-rolls the target if incompatible?

File: data/moves.ts [hunk 1]
```diff
@@ -11267,6 +11270,7 @@ export const Moves: {[moveid: string]: MoveData} = {
 			},
 			onRedirectTargetPriority: -1,
 			onRedirectTarget(target, source, source2) {
+				if (source.moveThisTurn !== 'mirrorcoat') return;
```
Comment (Zarel) [inline data/moves.ts:11273]:
```suggestion
			onRedirectTarget(target, source, source2, move) {
				if (move.id !== 'mirrorcoat') return;
```

File: data/moves.ts [hunk 2]
```diff
@@ -4275,6 +4276,8 @@ export const Moves: {[moveid: string]: MoveData} = {
 					// it failed
 					return false;
 				}
+				this.effectState.slot = target.getSlot();
+				this.effectState.user = target;
```
Comment (Zarel) [inline data/moves.ts:4280]:
```suggestion
```

File: data/moves.ts [hunk 3]
```diff
@@ -2645,6 +2645,7 @@ export const Moves: {[moveid: string]: MoveData} = {
 			},
 			onRedirectTargetPriority: -1,
 			onRedirectTarget(target, source, source2) {
+				if (source.moveThisTurn !== 'counter') return;
```
Comment (Zarel) [inline data/moves.ts:2648]:
```suggestion
			onRedirectTarget(target, source, source2, move) {
				if (move.id !== 'counter') return;
```

**Comment by pacmanboss256** [comment]
bump

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**